src/chains/ethereum/ethereum/tests/api/evm/evm.test.ts

```diff
@@ -132,6 +133,31 @@ describe("api", () => {
         const currentBlock = parseInt(await provider.send("eth_blockNumber"));
         assert.strictEqual(currentBlock, initialBlock + 1);
       });
+
+      it("should save the block before returning", async () => {
+        // slow down memdown's _batch function to consistently reproduce a past
+        // race condition where a block was mined and returned by evm_mine
+        // before it actually saved to the database. for history, the race
+        // condition issue is documented here:
+        // https://github.com/trufflesuite/ganache/issues/3060
+        memdown._batch = function (array, options, callback) {
+          setTimeout(() => {
+            memdown.prototype._batch(array, options, callback);
+          }, 20);
+        };
+        const options = { database: { db: memdown() } };
```

Comment:
Does this code work? I was thinking it should be more like (totally not tested!):

```javascript
        const db = memdown();
        let customBatchCalled = false;
        db._batch = (...args) => {
            setTimeout(() => {
              customBatchCalled = true;
              Reflect.apply(memdown.prototype._batch, db, args);
            }, 20);
          });
        };
        const options = { database: { db } };
        
        // ... other tests
        
        // make sure monkey patching _batch_ was actually used (protect against future changes in `memdown`):
        assert(customBatchCalled);
```

Also, I wonder how we can determine if 20ms is enough time to wait? :thinking:

Reply:
I've implemented your suggestion!

From my earlier comment: 
>I've created https://github.com/trufflesuite/ganache/pull/3102 that has these tests without the actual race condition fixes as a proof that the tests do what I say they should, and that my changes actually fix the bug.

So running this test with 20ms on a branch without the actual fix allowed us to consistently recreate this race condition. You'll recall that without the slowdown, the race condition causing an error in our tests was rare.

Reply:
Oh cool. That was fast!

I'm just wondering how we can _know_ 20ms is enough? I haven't given it deep consideration, and it's probably okay at 20ms.